plugins/alembic/module/vtkF3DAlembicReader.cxx

```diff
@@ -20,51 +22,334 @@
 #pragma warning(pop)
 #endif
 
+using IndicesContainer = std::vector<int32_t>;
+using V3fContainer = std::vector<Alembic::Abc::V3f>;
+using AttributesContainer = std::map<std::string, V3fContainer>;
+using PerFaceWavefrontIndicesTripletsContainer = std::vector<Alembic::Abc::V3i>;
+using PerMeshWavefrontIndicesTripletsContainer =
+  std::vector<PerFaceWavefrontIndicesTripletsContainer>;
+
+const size_t P_indices_offset = 0;
+const size_t uv_indices_offset = 1;
+const size_t N_indices_offset = 2;
+
+struct IntermediateGeometry
+{
+  AttributesContainer _attributes;
+  PerMeshWavefrontIndicesTripletsContainer _indices;
+  bool _uv_is_facevarying{false};
+  bool _N_is_facevarying{false};
+};
+
 class vtkF3DAlembicReader::vtkInternals
 {
-public:
-  vtkSmartPointer<vtkPolyData> ProcessIPolyMesh(const Alembic::AbcGeom::IPolyMesh& pmesh)
+  void SetupIndicesStorage(const Alembic::AbcGeom::Int32ArraySamplePtr& face_vertex_counts,
+    PerMeshWavefrontIndicesTripletsContainer& extracted_indices)
   {
-    vtkNew<vtkPoints> points;
-    vtkNew<vtkCellArray> polys;
-    vtkNew<vtkPolyData> polydata;
 
-    Alembic::AbcGeom::IPolyMeshSchema::Sample samp;
-    if (pmesh.getSchema().getNumSamples() > 0)
+    for (size_t i = 0; i < face_vertex_counts->size(); i++)
     {
-      pmesh.getSchema().get(samp);
+      auto this_face_vertex_count = face_vertex_counts->get()[i];
+      PerFaceWavefrontIndicesTripletsContainer per_face_indices(this_face_vertex_count);
+      extracted_indices.push_back(per_face_indices);
```

Comment:
Is it possible to avoid copies by calling something like that?

```suggestion
      extracted_indices.emplace_back({ face_vertex_counts->get()[i] });
```

Reply:
Done